src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+﻿using System;
+using System.Reactive.Disposables;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(
+        ":disabled",
+        ":secondaryButtonRight",
+        ":secondaryButtonSpan",
+        ":checkedFlyoutOpen",
+        ":flyoutOpen",
+        ":checkedTouchPressed",
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
+        ":primaryPressed",
+        ":primaryPointerOver",
+        ":secondaryPressed",
+        ":secondaryPointerOver")]
+    public class SplitButton : ContentControl, ICommandSource
+    {
+        /// <summary>
+        /// Raised when the user presses the primary part of the <see cref="SplitButton"/>.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> Click
+        {
+            add => AddHandler(ClickEvent, value);
+            remove => RemoveHandler(ClickEvent, value);
+        }
+
+        /// <summary>
+        /// Defines the <see cref="Click"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> ClickEvent =
+            RoutedEvent.Register<SplitButton, RoutedEventArgs>(
+                nameof(Click),
+                RoutingStrategies.Bubble);
```

Comment:
You don't need to register new routed event but use already registered instance from Button. 
Though it won't matter if you inherit control from Button.

Reply:
Can you confirm a code example? With my quick search of source code I don't think this can be done with RoutedEvent. At least every control is defining a new ClickEvent.

Reply:
Changes made in a commit. `public static readonly RoutedEvent<RoutedEventArgs> ClickEvent = Button.ClickEvent`